### PR TITLE
GDScript: Fix subclass script path issues

### LINKED ISSUE #67439: GDScript 2.0: Debugging inner classes is broken
### Godot version

v4.0.beta3.official [01ae26d31]

### System information

Windows 10

### Issue description

I want to debug my inner classes, but the dubuger skips the code.
Thus, it is currently not possible to debug inner classes.

### Steps to reproduce

Put this code into a new script file.
Set breakpoint at 
`clazz.test()` or `return "foo"`

Run debug and try to step into.

```
class TestInnerClass extends RefCounted:

	func test() -> String:
		return "foo"

func _ready():
	var clazz := TestInnerClass.new()
	clazz.test()
```

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by rune-scape**
Adds a method to GDScript to get the path of the script file it's in, as opposed to its resource path.
This is a refactor, so might fix unrelated bugs.
Fixes #67439

### CODE REVIEW

File: modules/gdscript/gdscript.cpp [hunk 1]
```diff
@@ -478,7 +478,7 @@ void GDScript::_clear_doc() {
 void GDScript::_update_doc() {
 	_clear_doc();
 
-	doc.script_path = "\"" + get_path().get_slice("://", 1) + "\"";
+	doc.script_path = "\"" + get_script_path().get_slice("://", 1) + "\"";
```
Comment (adamscott) [inline modules/gdscript/gdscript.cpp:481]: You could take the opportunity to lean out the code using `vformat()` for strings.

File: modules/gdscript/gdscript_analyzer.cpp [hunk 2]
```diff
@@ -3127,9 +3127,9 @@ void GDScriptAnalyzer::reduce_identifier(GDScriptParser::IdentifierNode *p_ident
 					Variant constant = GDScriptLanguage::get_singleton()->get_named_globals_map()[name];
 					Node *node = Object::cast_to<Node>(constant);
 					if (node != nullptr) {
-						Ref<Script> scr = node->get_script();
+						Ref<GDScript> scr = node->get_script();
 						if (scr.is_valid()) {
```
Comment (adamscott) [inline modules/gdscript/gdscript_analyzer.cpp:3131]: Nice catch.

**Review by adamscott — APPROVED** [review]
Works for me!

File: modules/gdscript/gdscript.cpp (hunk 1 above)
Comment (rune-scape) [reply modules/gdscript/gdscript.cpp]: good point!

**Review by Chaosus — APPROVED** [review]
Checked, works for me too.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**